openmc/kinetics/state.py

```diff
@@ -0,0 +1,1572 @@
+from abc import ABC
+from collections import OrderedDict
+import copy
+import sys
+
+import h5py
+import numpy as np
+import scipy.sparse as sps
+
+import openmc
+import openmc.kinetics
+import openmc.mgxs
+
+
+class State(ABC):
```

Comment:
My earlier comment stands -- although this class is intended to be abstract (via subclassing `ABC`), no methods are marked as abstract, which means any class could subclass `State` without having to implement specific methods. What is it that the subclasses (`OuterState`, `InnerState`) need to implement?